api/src/pcapi/core/offers/repository.py

```diff
@@ -162,7 +165,7 @@ def get_offers_by_filters(
                 <= period_ending_date
             )
 
-    return query.distinct(Offer.id)
+    return query
```

Comment:
On avait ajouté, puis supprimé, puis ré-ajouté ce distinct. Je me souviens de ce [commentaire de PR](https://github.com/pass-culture/pass-culture-api/pull/1607#discussion_r527698603), et de [cette PR de démonstration](https://github.com/pass-culture/pass-culture-api/pull/1613) d'un bug quand on enlève le distinct. Le distinct avait été ré-introduit ensuite avec [cette PR](https://github.com/pass-culture/pass-culture-api/pull/1614). Le code et les tests ont peut-être bougé, donc peut-être que ça n'est plus d'actualité...

Reply:
d'où ma question pour l'historique git.
En fait, le test ajouté avec le distinct a été supprimé dans le commit https://github.com/pass-culture/pass-culture-main/commit/b7db2042c
Je suis en train de voir s'il y a toujours besoin de ce distinct en ajoutant un test.

Reply:
Il faudra que je creuse plus, le test en question prend mal en compte les duplicata (SQLA fait du zèle et supprime les doublons mais avec la limite, c'est plus compliqué).

Reply:
Tu suspectes que retirer `distinct()` casse des choses ? Si c'est le cas, ne devrait-on pas le garder dans un premier temps ?

... Ou alors ton commit `unique offers for PRO with date filters` corrige le problème et fait en sorte qu'on n'a plus besoin de ce `distinct()` ?

J'ai apparemment loupé quelque chose, mais je ne sais pas quoi ! :)

Reply:
Le commit "unique offers for PRO with date filters" corrige le problème des occurrences multiples.
A priori, le seul point qui pourrait apporter des offres dupliquées, c'est le filtre par date. Une offre avec plusieurs stocks dans le créneau apparait plusieurs fois en SQL. SQLAlchemy fait le ménage après, mais je pense que ça risque de fausser la clause `LIMIT xxx`.
Je n'ai pas encore réussi à penser à un test permettant de vérifier/valider ça, je vais mouliner encore un peu voir si j'arrive à quelque chose.